Sources/MongoSwift/MongoCollection+Write.swift

```diff
@@ -290,60 +290,32 @@ public struct DeleteOptions: Encodable {
 public struct InsertOneResult {
     /// The identifier that was inserted. If the document doesn't have an identifier, this value
     /// will be generated and added to the document before insertion.
-    public let insertedId: BsonValue
+    public let insertedId: BsonValue?
 }
 
 /// The result of an `insertMany` command on a `MongoCollection`. 
 public struct InsertManyResult {
     /// Map of the index of the inserted document to the id of the inserted document.
-    public let insertedIds: [Int64: BsonValue]
-
-    /// Given an ordered array of insertedIds, creates a corresponding InsertManyResult.
-    internal init(fromArray arr: [BsonValue]) {
-        var inserted = [Int64: BsonValue]()
-        for (i, id) in arr.enumerated() {
-            let index = Int64(i)
-            inserted[index] = id
-        }
-        self.insertedIds = inserted
-    }
+    public let insertedIds: [BsonValue?]
```

Comment:
My guess is because `insertMany` is generally implemented as a bulk operation, and bulk operations can have multiple operations within. The map tells you which of these operations is associated with the inserted id. I don't feel very strongly which we way we go here, but @jmikola may feel differently.

Reply:
I vote for keeping this a map in accordance with the spec.

Consider a case where `insertMany` uses `ordered: false` and some documents in the middle of the requests fail. In that case, a map is necessary to know exactly which inserted IDs correspond to which documents in the original input.

Reply:
hmm, would we even get to that point? I think we would have already thrown an error because `mongoc_collection_insert_many` would return false. 
that said, I don't think we are correctly propagating writeErrors / writeConcernErrors in that case, and maybe we should also be sticking insertedIds / the `InsertManyResult` into the error?

Reply:
Correct to your last point. Exceptions for bulk writes (`insertMany()` and `bulkWrite()`) should include a write result, so it'd make sense to create the proper `insertedIds` map to inject therein.

I'll point out that while the [CRUD spec](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst) does say both of those methods throw a BulkWriteException, there is only one definition of that class in the spec and it includes a `writeResult` field as an optional BulkWriteResult. This is likely a failing of the spec, as it doesn't make sense for `insertMany` to throw something with the full result object from a `bulkWrite`. In that case, `writeResult` should be an InsertManyResult. Correcting this will likely require adding a new InsertManyException class to the spec. Mind creating a SPEC ticket for that with DCN: Maybe?

Reply:
I've created [SPEC-1147](https://jira.mongodb.org/browse/SPEC-1147), feel free to edit.

Reply:
also, I've gone back to using an `[Int: BsonValue?]` in `InsertManyResult`. I am also adding a note to throw a better error from here (and various other write methods) in SWIFT-139, once we've merged in your bulkWrite changes that include the structs like `WriteError` and `WriteConcernError` that we'll need to include in this error.